### DISCUSSION TITLE
Debye length when ionic strength = 0

### DISCUSSION THREAD

**Discussion by rwxayheee** [Electrostatics/apbs#182] [Q&A]
Dear APBS experts, 

I have a question on how to properly run an APBS calculation without ions. In my input file, I set the ion concentration to be 0. But from the APBS output, I found that the Debye length = 0 A. I'm wondering whether that means the screening is turned off, or maybe I have made some mistake in the input which leads to a Debye length of 0.. 

Below is my input: 
> read
>     mol pqr protein.pqr
> end
> elec
>     mg-auto
>     dime 513 513 513
> 
>     cglen 1000 1000 1000
>     fglen 256 256 256
> 
>     cgcent mol 1
>     fgcent mol 1
> 
>     mol 1
>     lpbe
>     bcfl sdh
> 
>     ion 1 0.00 1.62 # Na+ Born radius
>     ion -1 0.00 2.26 # Cl- Born radius
> 
>     pdie 4
>     sdie 78
>     srfm smol
> 
>     chgm spl2
> 
>     sdens 10.0
>     srad 1.40
>     swin 0.30
>     temp 298
>     calcenergy total
> 
>     calcforce no
>     write pot dx protein
> end
> quit

Also, I was wondering how diluted the solvent needs to be for the application of the screened coulomb potential. I was planning to calculate binding energy at different ionic strengths from 0 to 1M, but I'm worried whether 1M is valid or too high for the "mg-auto" type of calculation. 

Any comments/suggestions would be really appreciated. Thank you for your time and kind advice in advance! 

Thanks,
Amy

**Comment by sobolevnrm** [comment]
Hi @rwxayheee --

The [Debye length](https://en.wikipedia.org/wiki/Debye_length) isn't defined at zero ionic strength so I just had the output report "0".  This should probably be "infinite" or "undefined" instead.

In general, Poisson-Boltzmann theory does OK treating non-specific ion behavior at relatively low ionic strengths.  I get skeptical of PB results when any of the following conditions are met:

- There is a known specific-ion effect (e.g., fluoride behaves differently than chloride, potassium behaves differently than sodium, etc.) -- this is actually fairly common.
- There is a significant concentration of divalent (or higher valency) ions; e.g., greater than 50 mM
- The total ionic strength is greater than 400 mM

I don't have strong justifications for all of these conditions -- they're just rules of thumb that I use when applying PB methods to systems.  Caveat emptor.

Thanks,

Nathan

**Reply by rwxayheee** [reply]
Hi Nathan, 
Thank you so much for your kind reply. That is very helpful!!
